source/common/common/mem_block_builder.h

```diff
@@ -0,0 +1,136 @@
+#pragma once
```

Comment:
Can we split this out into a distinct PR for review? Thanks.

Reply:
done --> #9306